src/otx/core/metrics/accuracy.py

```diff
@@ -5,16 +5,198 @@
 
 from __future__ import annotations
 
-from typing import Callable, Sequence
+from typing import TYPE_CHECKING, Any, Callable, Literal, Sequence
 
 import torch
 from torch import nn
-from torchmetrics import Metric
-from torchmetrics.classification.accuracy import Accuracy, MultilabelAccuracy
+from torchmetrics import ConfusionMatrix, Metric
+from torchmetrics.classification.accuracy import Accuracy as TorchmetricAcc
+from torchmetrics.classification.accuracy import MultilabelAccuracy as TorchmetricMultilabelAcc
 
+if TYPE_CHECKING:
+    from torch import Tensor
 
-class HLabelAccuracy(Metric):
-    """Custom accuracy metric for h-label classification.
+    from otx.core.data.dataset.base import LabelInfo
+
+
+class NamedConfusionMatrix(nn.Module):
```

Comment:
Why don't you make it derived from `ConfusionMatrix`?

Reply:
I tried that way, however, I got the below error. I'm not sure why this happen, that's the reason why I selected this way.

```python3
class NamedConfusionMatrix(ConfusionMatrix):
    """Named Confusion Matrix to add row, col label names."""

    def __init__(
        self,
        task: str,
        num_classes: int,
        col_names: list[str],
        row_names: list[str],
    ):
        super().__init__(task=task, num_classes=num_classes)
        self.col_names = col_names
        self.row_names = row_names
```

ValueError: Unexpected keyword arguments: `col_names`, `row_names`

Reply:
```python
from __future__ import annotations
from typing import Any, Literal, Optional
from torchmetrics import ConfusionMatrix


class NamedConfusionMatrix(ConfusionMatrix):
    """Named Confusion Matrix to add row, col label names."""

    def __new__(
        cls,
        col_names: list[str],
        row_names: list[str],
        task: Literal["binary", "multiclass", "multilabel"],
        threshold: float = 0.5,
        num_classes: Optional[int] = None,
        num_labels: Optional[int] = None,
        normalize: Optional[Literal["true", "pred", "all", "none"]] = None,
        ignore_index: Optional[int] = None,
        validate_args: bool = True,
        **kwargs: Any,
    ) -> NamedConfusionMatrix:
        confusion_metric = super().__new__(
            cls,
            task=task,
            threshold=threshold,
            num_classes=num_classes,
            num_labels=num_labels,
            normalize=normalize,
            ignore_index=ignore_index,
            validate_args=validate_args,
            **kwargs,
        )

        confusion_metric.col_names = col_names
        confusion_metric.row_names = row_names
        return confusion_metric

    @property
    def col_names(self) -> list[str]:
        return self.col_names

    @property
    def row_names(self) -> list[str]:
        return self.row_names


matrix = NamedConfusionMatrix(["a", "b"], ["a", "b"], task="multiclass", num_classes=2)
print(matrix.col_names)
```

Reply:
Done, https://github.com/openvinotoolkit/training_extensions/pull/2994/commits/1b3ff689e2b28f59cc011324a952d5b405d88424